party3.tex

```diff
@@ -0,0 +1,33 @@
+
+
+Though the dev guide has a procedure \footnote{\url{https://developer.lsst.io/stack/packaging-third-party-eups-dependencies.html}}
+for making a third part package for inclusion in the pipelines distrib we have no official policy for how we accept and deal with such packages.
+
+
+
+There are several reasons to include third part code in our distributions:
+\begin{enumerate}
+\item There are packages which we use in our code e.g. starlink\_ast.  These are standard dependencies.  Currently these are put in lsst github org and probably should not be. Apart from that this is a well understood reason for third party code. \label{item:depend}
+\item There are packages which we want to use when working with the deployed software. AstroPy might be such an example - we don't need it to build or run our code but its sure handy for interaction with the data.\label{item:want}
```

Comment:
I suggest packages of this type should be explicitly *excluded* from distribution as part of the LSST Science Pipelines.

That doesn't mean we shouldn't provide some other means of installing them, making them available through the Science Platform, etc. But if I choose to install some piece of software and it “helpfully” installs some other stuff that I might want without my approval, that's just going to make me mad.

Reply:
Astropy is a bad example here because LSST Science Pipelines requires it and will break if astropy is missing.

Reply:
pandas is a better example for now.

Reply:
Put in pandas .. I think John we are back to product - we should not include these in our products (see section 2.1) but we may want them in some distribution especially if that is a docker container. If its a binary install the dependencies can take care of it.